### ISSUE TITLE
Arduino 1.5.5 Compatability issues

### ISSUE THREAD

**Issue by KurtE** [vanadiumlabs/arbotix#4]
First simple things:

There are some new lines that I believe need to be added to the boards.txt and programmers.txt.  You need to define what tool you are using. 

First for the Avrisp Serial programmer, mine now looks like:

```
avrispmkii.name=AVRISP mkII (serial)
avrispmkii.communication=serial
avrispmkii.protocol=stk500v2
avrispmkii.program.tool=avrdude
avrispmkii.program.extra_params=-P{serial.port}
```

Likewise I believe you need to define at least one more new line in boards.txt for upload tool, something like:

```

arbotix.name=ArbotiX
arbotix.upload.tool=avrdude
arbotix.upload.protocol=arduino
arbotix.upload.maximum_size=65536
arbotix.upload.speed=38400
arbotix.bootloader.low_fuses=0xFF
arbotix.bootloader.high_fuses=0xDC
arbotix.bootloader.extended_fuses=0xFD
arbotix.bootloader.file=atmega644p/ATmegaBOOT_644P.hex
arbotix.bootloader.unlock_bits=0x3F
arbotix.bootloader.lock_bits=0x0F
arbotix.build.mcu=atmega644p
arbotix.build.f_cpu=16000000L
arbotix.build.core=arbotix
arbotix.build.variant=standard
```

Need to do for each section.  Note: did not test this part as I was using programmer.  Also in the section I was doing, I also defined:
arbotixAd.upload.maximum_data_size=4096

Later may want to talk about using the main core code base and only define a new variant.  I believe we can either use the main Hardware Serial code or can get away with hacking up our own write function and use their receiver...  Have a version hacked up and testing now...  My section in the boards.txt looks like:

```
arbotixAd.name=ArbotiX Std

arbotixAd.upload.tool=avrdude
arbotixAd.upload.protocol=arduino
arbotixAd.upload.maximum_size=65536
arbotixAd.upload.maximum_data_size=4096
arbotixAd.upload.speed=38400

arbotixAd.bootloader.low_fuses=0xFF
arbotixAd.bootloader.high_fuses=0xDC
arbotixAd.bootloader.extended_fuses=0xFD
arbotixAd.bootloader.file=atmega644p/ATmegaBOOT_644P.hex
arbotixAd.bootloader.unlock_bits=0x3F
arbotixAd.bootloader.lock_bits=0x0F

arbotixAd.build.mcu=atmega644p
arbotixAd.build.f_cpu=16000000L
arbotixAd.build.core=arduino:arduino
arbotixAd.build.variant=serial1
```

Kurt

**Comment by mikeferguson** [comment]
Kurt, good to know. Are you currently working off the arduino-1.5.2 branch?

**Comment by KurtE** [comment]
Yes I am working with your 1.5.2 branch, and on Arduino running with  build 1.5.5 and daily builds.  I am also enlisted in one of the Arduino contributors branches: matthijskooijman/Arduino.  Found issues with using straight HardwareSerial code at 1mbs for AX12.  I debugged and reworked the ::write method to avoid the lock up in flush method.  I submitted an Issue on the main Arduino github account and found that it was a duplicate of one of Mattijs issues which he has a pull request in for.  

Have started doing some testing with my Phoenix code base using the HardwareSerial code and appears to work,  Will put up a version of my Phantom_Phoenix code up on my github account that uses this or conditionally simply implements my own write function and sync function. 

Note: I think the more recent changes to programmers.txt and boards.txt are a result of changes due to the support for Arduino Yun.

**Comment by mikeferguson** [comment]
Linking to http://forums.trossenrobotics.com/showthread.php?5786-Arbotix-on-Arduino-1.5(will-be-1.0.2)-IDE/page2 for reference

**Comment by KurtE** [comment]
Quick update: I will also post on Trossen forum thread as well.

Decided to try doing an upload using the serial port on the Arbotix instead of ISP, ran into a few more problems that I think I have working on my system.  Note: I have only mode the changes to the section of boards.txt for my logical board type that uses the default Arduino 1.5.x code base.

First I had to burn a bootloader on to the board.  Ran into issues.  Needed to define bootloader tool in boards.txt, like:

```
arbotixAd.bootloader.tool=avrdude
```

Then ran into issue that system was hard coding trying to load the bootloader from the Arduino install point not our hardware folder.  Found issue up on Arduino Issues: https://github.com/arduino/Arduino/issues/1176

So I updated platform.txt to the version out of Arduino 1.5.5 avr section.  I then changed this line from:

```
tools.avrdude.bootloader.pattern="{cmd.path}" "-C{config.path}" {bootloader.verbose} -p{build.mcu} -c{protocol} {program.extra_params} "-Uflash:w:{runtime.ide.path}/hardware/arduino/avr/bootloaders/{bootloader.file}:i" -Ulock:w:{bootloader.lock_bits}:m
```

To:

```
tools.avrdude.bootloader.pattern="{cmd.path}" "-C{config.path}" {bootloader.verbose} -p{build.mcu} -c{protocol} {program.extra_params} "-Uflash:w:{sketchbook.path}/hardware/arbotix/avr/bootloaders/{bootloader.file}:i" -Ulock:w:{bootloader.lock_bits}:m
```

After restarting IDE I was able to burn the bootloader.

I then connected up the comm port to the Arbotix and was able to download the program over it.  But again you must define the upload tool in boards.txt like:

```
arbotixAd.upload.tool=avrdude
```
